### PR TITLE
chore: add config to exclude certain entity types from default entity span enrichment

### PR DESCRIPTION
**Pull request by SrikarMannepalli**
## Description
This PR adds a Entity Span enricher config to allow for excluding certain entity types from default entity span enrichment

### Testing
Added unit tests.

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/EntitySpanEnricher.java [hunk 1]
```diff
@@ -11,12 +13,16 @@
 
 public class EntitySpanEnricher extends AbstractTraceEnricher {
   private static final Logger LOG = LoggerFactory.getLogger(EntitySpanEnricher.class);
+
+  private static final String ENTITY_TYPES_TO_BE_EXCLUDED_CONFIG_PATH = "entityTypesToBeExcluded";
```
Comment (skjindal93) [inline hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/EntitySpanEnricher.java:17]:
```suggestion
  private static final String ENTITY_TYPES_TO_BE_EXCLUDED_CONFIG_PATH = "excludeEntityTypes";
```

File: hypertrace-trace-enricher/trace-reader/src/main/java/org/hypertrace/trace/accessor/entities/DefaultTraceEntityAccessor.java [hunk 2]
```diff
@@ -54,13 +55,22 @@ class DefaultTraceEntityAccessor implements TraceEntityAccessor {
   }
 
   @Override
-  public void writeAssociatedEntitiesForSpanEventually(StructuredTrace trace, Event span) {
+  public void writeAssociatedEntitiesForSpanEventually(
```
[the remark this answers is not in the document]
Comment (SrikarMannepalli) [reply hypertrace-trace-enricher/trace-reader/src/main/java/org/hypertrace/trace/accessor/entities/DefaultTraceEntityAccessor.java:58]: `DefaultClientRegistry` takes in the client config only. Wondering if putting this config in clients would be ideal? Or should we update the client registry to take the entire config? Bit split on what would be better 🤔  
Screenshot 2023-09-07 at 2 00 36 PM

[the remark this answers is not in the document]
Comment (aaron-steinfeld) [reply hypertrace-trace-enricher/trace-reader/src/main/java/org/hypertrace/trace/accessor/entities/DefaultTraceEntityAccessor.java:58]: I would put it in the client config. It's basically the same scope as `trace.entity.write.throttle.duration` right? so use that as a template, maybe trace.entity.write.excludedTypes or something.

File: hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/clients/DefaultClientRegistry.java [hunk 3]
```diff
@@ -86,6 +91,10 @@ public DefaultClientRegistry(
                 config.hasPath(TRACE_ENTITY_WRITE_THROTTLE_DURATION)
                     ? config.getDuration(TRACE_ENTITY_WRITE_THROTTLE_DURATION)
                     : Duration.ofSeconds(15))
+            .withExcludeEntityTypes(
+                config.hasPath(TRACE_ENTITY_WRITE_EXCLUDED_ENTITY_TYPES)
+                    ? new HashSet<>(config.getStringList(TRACE_ENTITY_WRITE_EXCLUDED_ENTITY_TYPES))
```
Comment (aaron-steinfeld) [inline hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/clients/DefaultClientRegistry.java:96]: nit. Set.copyOf (no mutable config)

File: hypertrace-trace-enricher/trace-reader/src/main/java/org/hypertrace/trace/accessor/entities/TraceEntityAccessorBuilder.java [hunk 4]
```diff
@@ -11,6 +13,7 @@ public class TraceEntityAccessorBuilder {
   private final EntityDataClient entityDataClient;
   private final CachingAttributeClient attributeClient;
   private Duration entityWriteThrottleDuration = Duration.ofSeconds(15);
+  private Set<String> excludedEntityTypes = new HashSet<>();
```
Comment (aaron-steinfeld) [inline hypertrace-trace-enricher/trace-reader/src/main/java/org/hypertrace/trace/accessor/entities/TraceEntityAccessorBuilder.java:16]: emptyset or similar. builder data shouldn't be mutable except through builder interfaces

**Review by aaron-steinfeld — DISMISSED** [review]

**Comment by aaron-steinfeld** [comment]
Sorry just noticed dep check is failing. It should be required here, not sure why it's not. going to update that, can you take a look?

**Review by kotharironak — APPROVED** [review]
lgtm

**Pull request merged**